File: leetcode/125_valid_palindrome/main.py
def is_palindrome(s: str) -> bool:
    s = s.lower()
    
    left = 0
    right = len(s)-1

    while left < right:
        if not s[left].isalnum():
            left += 1
            continue
        if not s[right].isalnum():
            right -= 1
            continue
        
        if s[left] != s[right]:
            return False
        
        left += 1
        right -= 1

    return True


print(is_palindrome("roma"))    
print(is_palindrome("A man, a plan, a canal: Panama"))    
print(is_palindrome("race a car"))    
print(is_palindrome(""))    
print(is_palindrome(" "))    

File: leetcode/125_valid_palindrome/main.go
package main

import (
	"fmt"
	"strings"
	"unicode"
)

func main() {
	fmt.Println(isPalindrome("roma"))
	fmt.Println(isPalindrome(""))
	fmt.Println(isPalindrome(" "))
	fmt.Println(isPalindrome("A man, a plan, a canal: Panama"))
	fmt.Println(isPalindrome("race a car"))
}

func isPalindrome(s string) bool {
	ss := []rune(s)
	
	left := 0
	right := len(s)-1

	for left < right {

		if !unicode.IsLetter(ss[left]) {
			left++
			continue
		}

		if !unicode.IsLetter(ss[right]) {
			right--
			continue
		}

		if strings.ToLower(string(ss[left])) != strings.ToLower(string(ss[right])) {
			return false
		}

		left++
		right--
	}

	return true
}

